spring-integration-core/src/main/java/org/springframework/integration/aggregator/AbstractCorrelatingMessageHandler.java

```diff
@@ -561,6 +561,7 @@ protected void handleMessageInternal(Message<?> message) {
 		this.logger.debug(() -> "Handling message with correlationKey [" + correlationKey + "]: " + message);
 
 		UUID groupIdUuid = UUIDConverter.getUUID(correlationKey);
+		Assert.notNull(groupIdUuid, "groupIdUuid must not be null.");
```

Comment:
Same question here: if `correlationKey` is not null (according to the `Assert.state(correlationKey != null)`), then we don't need this extra check here.